pkg/trace/schema/consensus.go

```diff
@@ -21,162 +21,166 @@ func ConsensusTables() []string {
 // Schema constants for the consensus round state tracing database.
 const (
 	// RoundStateTable is the name of the table that stores the consensus
-	// state traces. Follows this schema:
-	//
-	// | time | height | round | step |
+	// state traces.
 	RoundStateTable = "consensus_round_state"
-
-	// StepFieldKey is the name of the field that stores the consensus step. The
-	// value is a string.
-	StepFieldKey = "step"
 )
 
+// RoundState describes schema for the "consensus_round_state" table.
+type RoundState struct {
+	Height int64                 `json:"height"`
+	Round  int32                 `json:"round"`
+	Step   cstypes.RoundStepType `json:"step"`
+}
+
+// Table returns the table name for the RoundState struct.
+func (r RoundState) Table() string {
+	return RoundStateTable
+}
+
+// InfluxRepr returns the InfluxDB representation of the RoundState struct.
+func (r RoundState) InfluxRepr() (map[string]interface{}, error) {
+	return toMap(r)
+}
+
 // WriteRoundState writes a tracing point for a tx using the predetermined
-// schema for consensus state tracing. This is used to create a table in the following
-// schema:
-//
-// | time | height | round | step |
-func WriteRoundState(client *trace.Client, height int64, round int32, step cstypes.RoundStepType) {
-	client.WritePoint(RoundStateTable, map[string]interface{}{
-		HeightFieldKey: height,
-		RoundFieldKey:  round,
-		StepFieldKey:   step.String(),
-	})
+// schema for consensus state tracing.
+func WriteRoundState(client trace.Tracer, height int64, round int32, step cstypes.RoundStepType) {
+	client.Write(RoundState{Height: height, Round: round, Step: step})
 }
 
 // Schema constants for the "consensus_block_parts" table.
 const (
 	// BlockPartsTable is the name of the table that stores the consensus block
 	// parts.
-	// following schema:
-	//
-	// | time | height | round | index | peer | transfer type |
 	BlockPartsTable = "consensus_block_parts"
-
-	// BlockPartIndexFieldKey is the name of the field that stores the block
-	// part
-	BlockPartIndexFieldKey = "index"
 )
 
+// BlockPart describes schema for the "consensus_block_parts" table.
+type BlockPart struct {
+	Height       int64        `json:"height"`
+	Round        int32        `json:"round"`
+	Index        int32        `json:"index"`
+	Peer         p2p.ID       `json:"peer"`
+	TransferType TransferType `json:"transfer_type"`
+}
+
+// Table returns the table name for the BlockPart struct.
+func (b BlockPart) Table() string {
+	return BlockPartsTable
+}
+
+// InfluxRepr returns the InfluxDB representation of the BlockPart struct.
+func (b BlockPart) InfluxRepr() (map[string]interface{}, error) {
+	return toMap(b)
+}
```

Comment:
initially was not planning on removing influx support, but using reflection to convert to map of interfaces seems so silly. We could have kept the huge number of constants, but they are redundant to the struct definitions which is equally bad.